File: 0011-container-with-most-water/0011-container-with-most-water.c
int maxArea(int* height, int heightSize) {
    
    int left=0;
    int right=heightSize-1;
    int maxarea=0;
    
    while(left<right){
        int area=(height[left]<height[right] ? height[left] : height[right]) * (right-left);
        maxarea=(area>maxarea) ? area : maxarea;
        
        if(height[left]<height[right]){
            left++;
        }
        else{
            right--;
        }
    }
    return maxarea;
}

File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int left=0;
        int right=height.length-1;
        int res=0;
        
        while(left < right){
            int length=Math.min(height[left],height[right]);
            int breadth=(right-left);
            int area=length*breadth;
            res=Math.max(area,res);
            
            if(height[left]<height[right]){
                left++;
            }
            else{
                right--;
            }
        }
        return res;
    }
}